client/src/components/Markdown/Elements/HistoryDatasetDisplay.vue

```diff
@@ -33,14 +33,41 @@
                 </span>
             </b-card-header>
             <b-card-body>
-                <LoadingSpan v-if="loading" message="Loading Dataset" />
-                <div v-else class="embedded-dataset content-height">
-                    <pre v-if="itemContent.item_data">
-                        <code class="text-normalwrap">{{ itemContent.item_data }}</code>
-                    </pre>
-                    <div v-else>No content found.</div>
-                    <b-link v-if="itemContent.truncated" :href="itemContent.item_url"> Show More... </b-link>
-                </div>
+                <UrlDataProvider :url="itemUrl" v-slot="{ result: itemContent, loading, error }">
+                    <LoadingSpan v-if="loading" message="Loading Dataset" />
+                    <div v-else-if="error">{{ error }}</div>
+                    <div v-else class="embedded-dataset content-height">
+                        <b-embed
+                            v-if="['pdf', 'html'].includes(datasetType)"
+                            type="iframe"
+                            aspect="16by9"
+                            :src="displayUrl"
+                        />
+                        <div v-else-if="itemContent.item_data">
+                            <div v-if="datasetType == 'tabular'">
```

Comment:
I think we have some abstraction to deal with datatypes in a more uniform way, that will also work for subclasses with getDatatypesMapper / https://github.com/mvdbeek/galaxy/blob/611fb8fe8b3a1994e97d7399a6aa8202bffe7a77/client/src/components/Datatypes/model.js#L7.

Reply:
Added datatype mapping to match the `tabular` datatype, so the table will now also be shown for example 'bed' files.